rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,514 @@
+        Both of these look for the same entities based on the same training data
+        leading to ambiguity in the results.
+
+        Args:
+           node_names: names of all components
+        """
+        extractors_in_configuration: Set[
+            Type[GraphComponent]
+        ] = self._component_types.intersection(TRAINABLE_EXTRACTORS)
+        if len(extractors_in_configuration) > 1:
+            rasa.shared.utils.io.raise_warning(
+                f"You have defined multiple entity extractors that do the same job "
+                f"in your configuration: "
+                f"{types_to_str(extractors_in_configuration)}. "
+                f"This can lead to the same entity getting "
+                f"extracted multiple times. Please read the documentation section "
+                f"on entity extractors to make sure you understand the implications: "
+                f"{DOCS_URL_COMPONENTS}#entity-extractors"
+            )
+
+    def _warn_of_competition_with_regex_extractor(
+        self, training_data: TrainingData
+    ) -> None:
+        """Warns when regex entity extractor is competing with a general one.
+
+        This might be the case when the following conditions are all met:
+        * You are using a general entity extractor and the `RegexEntityExtractor`
+        * AND you have regex patterns for entity type A
+        * AND you have annotated text examples for entity type A
+
+        Args:
+            training_data: the training data
+        """
+        present_general_extractors = self._component_types.intersection(
+            TRAINABLE_EXTRACTORS
+        )
+        has_general_extractors = len(present_general_extractors) > 0
+        has_regex_extractor = (
+            RegexEntityExtractorGraphComponent in self._component_types
+        )
+
+        regex_entity_types = {rf["name"] for rf in training_data.regex_features}
+        overlap_between_types = training_data.entities.intersection(regex_entity_types)
+        has_overlap = len(overlap_between_types) > 0
+
+        if has_general_extractors and has_regex_extractor and has_overlap:
+            rasa.shared.utils.io.raise_warning(
+                f"You have an overlap between the "
+                f"{RegexEntityExtractorGraphComponent.__name__} and the "
+                f"statistical entity extractors "
+                f"{types_to_str(present_general_extractors)} "
+                f"in your configuration. Specifically both types of extractors will "
+                f"attempt to extract entities of the types "
+                f"{', '.join(overlap_between_types)}. "
+                f"This can lead to multiple "
+                f"extraction of entities. Please read "
+                f"{RegexEntityExtractorGraphComponent.__name__}'s "
+                f"documentation section to make sure you understand the "
+                f"implications: {DOCS_URL_COMPONENTS}#regexentityextractor"
```

Comment:
```suggestion
                f"'{RegexEntityExtractorGraphComponent.__name__}' and the "
                f"statistical entity extractors "
                f"{types_to_str(present_general_extractors)} "
                f"in your configuration. Specifically both types of extractors will "
                f"attempt to extract entities of the types "
                f"{', '.join(overlap_between_types)}. "
                f"This can lead to multiple "
                f"extraction of entities. Please read "
                f"'{RegexEntityExtractorGraphComponent.__name__}'s' "
                f"documentation section to make sure you understand the "
                f"implications:", docs=f"{DOCS_URL_COMPONENTS}#regexentityextractor"
```